src/devices/src/virtio/net/event_handler.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::virtio::net::device::Net;
+use crate::virtio::{VirtioDevice, RX_INDEX, TX_INDEX};
+use logger::{Metric, METRICS};
+use polly::event_manager::EventHandler;
+use polly::pollable::{Pollable, PollableOp, PollableOpBuilder};
+use std::os::unix::io::AsRawFd;
+
+impl EventHandler for Net {
+    fn handle_read(&mut self, source: Pollable) -> Vec<PollableOp> {
+        if !self.is_activated() {
+            warn!("The device is not yet activated. Events can not be handled.");
+            return vec![];
+        }
+
+        let virtq_rx_ev_fd = self.queue_evts[RX_INDEX].as_raw_fd();
+        let virtq_tx_ev_fd = self.queue_evts[TX_INDEX].as_raw_fd();
+        let rx_rate_limiter_fd = self.rx_rate_limiter.as_raw_fd();
+        let tx_rate_limiter_fd = self.tx_rate_limiter.as_raw_fd();
+        let tap_fd = self.tap.as_raw_fd();
+
+        match source {
+            _ if source == virtq_rx_ev_fd => self.process_rx_queue_event(),
+            _ if source == tap_fd => self.process_tap_rx_event(),
+            _ if source == virtq_tx_ev_fd => self.process_tx_queue_event(),
+            _ if source == rx_rate_limiter_fd => self.process_rx_rate_limiter_event(),
+            _ if source == tx_rate_limiter_fd => self.process_tx_rate_limiter_event(),
+            _ => {
+                error!("Unknown event source.");
+                METRICS.net.event_fails.inc();
+            }
+        }
+
+        vec![]
+    }
+
+    fn init(&self) -> Vec<PollableOp> {
```

Comment:
nit: testing this `init` function is fairly simple. See `src/firecracker/src/metrics.rs` for an example.